File: 2018/day06_2.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any

type Coords = tuple[int, int]


def solution(content: list[str]) -> Any:
    pivots = [parse(line) for line in content]
    hlimit = max(col for col, _ in pivots)
    vlimit = max(row for _, row in pivots)
    the_map = [
        sum(manhattan((x, y), pivot) for pivot in pivots)
        for x in range(hlimit + 1)
        for y in range(vlimit + 1)
    ]
    return len([c for c in the_map if c < 10_000])


def manhattan(c1: Coords, c2: Coords):
    x1, y1 = c1
    x2, y2 = c2
    return abs(x1 - x2) + abs(y1 - y2)


def parse(line: str) -> Coords:
    [x, y] = line.split(", ")
    return (int(x), int(y))


def main() -> None:
    with open("day06.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()

File: 2018/day06-2.ts
type Coords = {
  x: number;
  y: number;
};

function solution(content: string) {
  const pivots = content.split("\n").map(parse);
  const hlimit = Math.max(...pivots.map(({ x }) => x));
  const vlimit = Math.max(...pivots.map(({ y }) => y));
  const map = Array.from({ length: hlimit + 1 })
    .map((_, x) =>
      Array.from({ length: vlimit + 1 }).map((_, y) =>
        pivots.reduce((acc, cur) => acc + manhattan({ x, y }, cur), 0)
      )
    )
    .flat();
  return map.filter((item) => item < 10_000).length;
}

function manhattan(c1: Coords, c2: Coords) {
  return Math.abs(c1.x - c2.x) + Math.abs(c1.y - c2.y);
}

function parse(line: string): Coords {
  const [x, y] = line.split(", ").map(Number);
  return { x, y };
}

Deno.readTextFile("day06.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
